internal/predicate/nonadminbackup_predicate.go

```diff
@@ -19,86 +19,40 @@ package predicate
-type NonAdminBackupPredicate struct {
-	Logger logr.Logger
-}
+const nonAdminBackupPredicateKey = "NonAdminBackupPredicate"
 
-func getNonAdminBackupPredicateLogger(ctx context.Context, name, namespace string) logr.Logger {
-	return log.FromContext(ctx).WithValues("NonAdminBackupPredicate", types.NamespacedName{Name: name, Namespace: namespace})
-}
+// NonAdminBackupPredicate contains event filters for Non Admin Backup objects
+type NonAdminBackupPredicate struct{}
 
 // Create event filter
 func (NonAdminBackupPredicate) Create(ctx context.Context, evt event.CreateEvent) bool {
-	namespace := evt.Object.GetNamespace()
-	name := evt.Object.GetName()
-	logger := getNonAdminBackupPredicateLogger(ctx, name, namespace)
-	logger.V(1).Info("NonAdminBackupPredicate: Received Create event")
-	if nonAdminBackup, ok := evt.Object.(*nacv1alpha1.NonAdminBackup); ok {
-		if nonAdminBackup.Status.Phase == constant.EmptyString || nonAdminBackup.Status.Phase == nacv1alpha1.NonAdminBackupPhaseNew {
-			logger.V(1).Info("NonAdminBackupPredicate: Accepted Create event")
-			return true
-		}
-	}
-	logger.V(1).Info("NonAdminBackupPredicate: Rejecting Create event")
-	return false
+	logger := function.GetLogger(ctx, evt.Object, nonAdminBackupPredicateKey)
+	logger.V(1).Info("Accepted Create event")
+	return true
 }
 
 // Update event filter
 func (NonAdminBackupPredicate) Update(ctx context.Context, evt event.UpdateEvent) bool {
-	namespace := evt.ObjectNew.GetNamespace()
-	name := evt.ObjectNew.GetName()
-	logger := getNonAdminBackupPredicateLogger(ctx, name, namespace)
-	logger.V(1).Info("NonAdminBackupPredicate: Received Update event")
+	logger := function.GetLogger(ctx, evt.ObjectNew, nonAdminBackupPredicateKey)
 
+	// spec change
 	if evt.ObjectNew.GetGeneration() != evt.ObjectOld.GetGeneration() {
-		logger.V(1).Info("NonAdminBackupPredicate: Accepted Update event - generation change")
+		logger.V(1).Info("Accepted Update event")
 		return true
 	}
 
-	if nonAdminBackupOld, ok := evt.ObjectOld.(*nacv1alpha1.NonAdminBackup); ok {
-		if nonAdminBackupNew, ok := evt.ObjectNew.(*nacv1alpha1.NonAdminBackup); ok {
-			oldPhase := nonAdminBackupOld.Status.Phase
-			newPhase := nonAdminBackupNew.Status.Phase
-
-			// New phase set, reconcile
-			if oldPhase == constant.EmptyString && newPhase != constant.EmptyString {
-				logger.V(1).Info("NonAdminBackupPredicate: Accepted Update event - phase change")
-				return true
-			} else if oldPhase == nacv1alpha1.NonAdminBackupPhaseNew && newPhase == nacv1alpha1.NonAdminBackupPhaseCreated {
-				logger.V(1).Info("NonAdminBackupPredicate: Accepted Update event - phase created")
-				return true
-			}
-		}
-	}
-	logger.V(1).Info("NonAdminBackupPredicate: Rejecting Update event")
-
+	logger.V(1).Info("Rejected Update event")
 	return false
 }
 
 // Delete event filter
 func (NonAdminBackupPredicate) Delete(ctx context.Context, evt event.DeleteEvent) bool {
```

Comment:
NIT: `func (NonAdminBackupPredicate) <-> func (n NonAdminBackupPredicate)`

Reply:
this is wrong

you are creating a variable (`n`), that is not used. Our linters prevent that

Reply:
I would not term it as wrong because the linter is preventing it. Its just Go convention. Had you had any field in the `NonAdminBackupPredicate` that we are using in the funcs then the linter would not have complained. Anyways, I am fine either way.

Reply:
It's up to us to decide, we can always use on that particular line:
https://golangci-lint.run/usage/false-positives/#nolint-directive